src/clicast.ts

```diff
@@ -0,0 +1,264 @@
+      vectorMessage(`   could NOT find '${vcastqtName}' here: ${vcastInstallationPath}`);
+    }
+  }
+  else {
+    vectorMessage(`   could NOT find '${clicastName}' here: ${vcastInstallationPath}`);
+  }
+  return toolsFound;
+}
+
+
+function convertTestScriptContents(scriptPath: string) {
+  // Read the file
+  let originalLines = fs.readFileSync(scriptPath).toString().split(os.EOL);
+  let newLines: string[] = [];
+
+  // Modify the lines
+  for (let line of originalLines) {
+    if (line == "TEST.NEW") line = "TEST.REPLACE";
+    newLines.push(line);
+  }
+
+  // Join the modified lines back into a single string
+  const modifiedContent = newLines.join("\n");
+
+  // Write the modified content back to the file
+  fs.writeFileSync(scriptPath, modifiedContent, "utf8");
+}
+
+export async function openTestScript(nodeID: string) {
+  // this can get called for a unit, environment, function, or test
+
+  const testNode: testNodeType = getTestNode(nodeID);
+
+  const enclosingDirectory = path.dirname(testNode.enviroPath);
+  const scriptPath = testNode.enviroPath + ".tst";
+
+  let commandToRun: string = `${clicastCommandToUse} ${getClicastArgsFromTestNode(
+    testNode
+  )} test script create ${scriptPath}`;
+  const commandStatus = executeCommand(commandToRun, enclosingDirectory);
+  if (commandStatus.errorCode == 0) {
+    // Improvement needed:
+    // It would be nice if vcast generated the scripts with TEST.REPLACE, but for now
+    // convert TEST.NEW to TEST.REPLACE so doing an "immediate load" works without error
+    convertTestScriptContents(scriptPath);
+
+    // open the script file for editing
+    vscode.workspace.openTextDocument(scriptPath).then(
+      (doc: vscode.TextDocument) => {
+        vscode.window.showTextDocument(doc);
+      },
+      (error: any) => {
+        vectorMessage(error);
+      }
+    );
+  }
+}
+
+
+async function adjustScriptContentsBeforeLoad(scriptPath: string) {
+
+  // There are some things that need updating before we can load the 
+  // script into VectorCAST:
+  //   - The requirement key lines need to be split into two lines
+  //     We insert lines like TEST.REQUIREMENT_KEY: key | description,
+  //     but VectorCAST only allows the key, so we turn the description
+  //     into a comment.
+  //
+  //   - <might be more things to do later>
+
+  let originalLines = fs.readFileSync(scriptPath).toString().split("\n");
+  let newLines: string[] = [];
+  for (let line of originalLines) {
+    if (line.startsWith("TEST.REQUIREMENT_KEY:")) {
+      const keyLineParts = line.split("|");
+      if (keyLineParts.length == 2) {
+        newLines.push("-- Requirement Title: " + keyLineParts[1]);
+        newLines.push(keyLineParts[0].trim());
+      }
+      else {
+        newLines.push(line);
+      }
+    }
+    else {
+      newLines.push(line);
+    }
+  }
+  fs.writeFileSync(scriptPath, newLines.join("\n"), "utf8");
+}
+
+
+const url = require("url");
+export async function loadTestScript() {
```

Comment:
We have `loadTestScript()`, `deleteTest()` and others that represent testPane interactions. We also have "low-level" `clicast` - related functions like `executeClicastCommand()`. Maybe we could leave the "low-level stuff" in this file and move the "test pane interactions" to a separate file (maybe there are some other interaction functions still in `testPane.ts` at the moment?)

Reply:
Ok I will use the convention that the commands are in testPane, and the clicast related logic is in vcastUtilities.

Let me know what you think